File: frontend/src/app/models/review.ts
export interface Review {
    id: number;
    userId: number;
    userName: string;
    productId: number;
    rating: number; // 1-5 stars
    comment: string;
    createdAt: string;
    updatedAt: string;
}

export interface ReviewStats {
    averageRating: number;
    reviewCount: number;
    ratingDistribution: Array<{
        rating: number;
        count: number;
    }>;
}

File: backend-microservices/product-service/src/main/java/com/project/productservice/models/Review.java
package com.project.productservice.models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "reviews")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Review extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // user thuoc user-service -> chi luu id + ten snapshot
    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "user_name", length = 100)
    private String userName;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "rating", nullable = false)
    private Integer rating; // 1-5 stars

    @Column(name = "comment", columnDefinition = "TEXT")
    private String comment;
}

